x-pack/plugins/apm/server/routes/alerts/rule_types/utils/get_interval_in_seconds.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { INTERVAL_MULTIPLIER_FOR_LOOKBACK } from '../../../../../common/rules/apm_rule_types';
+
+const intervalUnits = ['y', 'M', 'w', 'd', 'h', 'm', 's', 'ms'];
+const INTERVAL_STRING_RE = new RegExp(
+  '^([0-9\\.]*)\\s*(' + intervalUnits.join('|') + ')$'
+);
+
+interface UnitsToSeconds {
+  [unit: string]: number;
+}
+
+const units: UnitsToSeconds = {
+  ms: 0.001,
+  s: 1,
+  m: 60,
+  h: 3600,
+  d: 86400,
+  w: 86400 * 7,
+  M: 86400 * 30,
+  y: 86400 * 356,
+};
+
+export const getIntervalInSeconds = (interval: string): number => {
+  const matches = interval.match(INTERVAL_STRING_RE);
+  if (matches) {
+    return (
+      parseFloat(matches[1]) *
+      INTERVAL_MULTIPLIER_FOR_LOOKBACK *
```

Comment:
This looks wrong. `INTERVAL_MULTIPLIER_FOR_LOOKBACK` looks APM specific. However, the rest of this code should be completely APM agnostic (even Elastic agnostic). It should simply convert `"5m"` into `300`.

Why is `INTERVAL_MULTIPLIER_FOR_LOOKBACK` used here?

Reply:
It will convert `5m` to `1500`. I have added unit tests for this function in `x-pack/plugins/apm/server/routes/alerts/rule_types/utils/get_interval_in_seconds.test.ts` for reference.

If the rule is configured to lookback `5m`, the chart will show wider range of data by multiplying the interval. So for `5m`, it will show last `25 minutes` of data, for `3h`, showing last `15 hours` of data and so on.

Reply:
removed `getIntervalInSeconds`